tests/test_common.h

```diff
@@ -0,0 +1,93 @@
+/**
+ * @file test_common.h
+ * @author Irfan
+ * @brief common header file for all tests to facilitate uniform logging format
+ *
+ * @copyright
+ *
+ * This source code is licensed under BSD 3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#ifndef SR_TEST_COMMON_H
+#define SR_TEST_COMMON_H
+
+#include <pthread.h>
+#include <assert.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include "sysrepo.h"
+#include "tests/config.h"
+
+/**
+ * Only function that needs to be called from test code
+ * Initializes callback and facilitates logging to stderr
+ * with timestamps and thread-id
+ */
+static void test_log_init();
+
+#define LOG_INF(...) _test_log(SR_LL_INF, __VA_ARGS__)
+#define LOG_ERR(...) _test_log(SR_LL_ERR, __VA_ARGS__)
+#define LOG_WRN(...) _test_log(SR_LL_WRN, __VA_ARGS__)
+
+static void
+_test_log_cb(sr_log_level_t level, const char *message);
+
+static void _test_log(sr_log_level_t ll, ...);
+
+static void test_log_init()
```

Comment:
Please no code in headers, I do not have good experiences with that. Also, is anything except the logging callback needed? Tests themselves do not log anything and I do not think it is needed.

Reply:
As we expand the UT, and features. It will be good to add some logs to the tests themselves to easily understand what's being done, when a failure happens.

Reply:
Alright, I suppose, but then name the macros something test-specific, `TLOG_ERR` at least, for example.